src/hotspot/share/code/debugInfo.hpp

```diff
@@ -165,6 +187,63 @@ class ObjectValue: public ScopeValue {
   void print_fields_on(outputStream* st) const;
 };
 
+// An ObjectMergeValue describes objects that were inputs to a Phi in C2 and at
+// least one of them was scalar replaced.
+// '_selector' is an integer value that will be '-1' if during the execution of
+// the C2 compiled code the path taken was that of the Phi input that was NOT
+// scalar replaced. In that case '_merge_pointer' is a pointer to an already
+// allocated object. If '_selector' is not '-1' then it should be the index
+// of a candidate object in '_possible_objects'. That candidate object is an
+// ObjectValue describing an object that was scalar replaced.
+
+class ObjectMergeValue: public ScopeValue {
```

Comment:
Why you did not make subclass of ObjectValue? You would need to check `sv->is_object_merge()` first before `sv->is_object()` in few places. But on other hand you don't need to duplicates ObjectValue`s fields and asserts.

Reply:
Let me try that and see how it looks.

Reply:
Hi @vnkozlov, just FYI. I made the changes that you suggested. Please let me know what you think.